docs/android/tips/leak_canary.md

```diff
@@ -0,0 +1,42 @@
+---
+title: "Leak Canary 🐤"
+sidebar_label: "Leak Canary"
+---
+
+## How to Disable Leak Canary in Debug Builds
+
+[LeakCanary](https://square.github.io/leakcanary/) is a powerful tool for detecting memory leaks in Android applications. However, there are scenarios where you might want to disable it, such as when preparing a debug build for performance testing or when it's not needed.
+
+### Disabling LeakCanary via Gradle Command
+
+You can disable LeakCanary manually by passing the `-PnoLeakCanary` flag in the Gradle command. For example:
+
+```bash
+./gradlew app:assembleFullDebug -PnoLeakCanary
+```
+This flag ensures that LeakCanary is excluded from the build.
+
+### Disabling LeakCanary via properties file
+
+Alternatively, you can disable LeakCanary by setting the noLeakCanary property in the gradle.properties file. This can be done at either the project level or the home level.
+
+```properties
+noLeakCanary=true
+```
+
+::::warning
+If it is set you need to update the lockfile otherwise gradle complains about the deps being not used.
+It can be done with
+
+```bash
+./gradlew alldependencies --write-locks
+```
+
+::::
+
+## Best Practices for Using LeakCanary
+
+- **Enable Only in Debug Builds:** LeakCanary should only be included in debug builds to avoid impacting performance in production builds.
+- **Regularly Monitor Memory Leaks:** Use LeakCanary during development to identify and fix memory leaks early.
+- **Document Known Leaks:** If a memory leak is caused by a third-party library and cannot be fixed immediately, document it for future reference.
+- **Speak about it:** If a leak is reported don't hesitate to get in touch with other contributors on Discord about it.
```

Comment:
Don't we have issues for this?

Reply:
Well while developing because can get notified by leak canary and they don't know what to do with it. It's not super explicit sometimes. I'm suggesting here to get in touch with the community when it happens.

Reply:
But then:
1. if the leak is in released code, it should be an issue
2. if the leak in in progress code, they should create a thread either in the support channels on discord. (or head into the android project, but the support thread exists for this use case)

But I think we should be more clear in that, because now it reads like "feel free to PM any contributor for help"

Reply:
For leaks it's kinda hard to tracks them outside of development environment. For instance we don't have Leak Canary in prod even beta (it's on purpose) so nothing is reporting leaks.

And it doesn't happen very often since we are using a modern language, so most probably only a dev working on his PR is going to find the leak. And I would prefer to PM me about it rather than ignoring it TBH.

Reply:
But PMing you isn't scalable. Even though you might be the one picking it up, you rather want someone to create a thread so others can also find it using search for example. This way anyone who knows how this works can pick it up, rather than it happens behind closed doors

Reply:
> if the leak in in progress code, they should create a thread either in the support channels on discord

In cases where "in progress" means merged but not yet released as a stable version, an issue would be fine as well (and preferred in my opinion, easier to keep track of/can be linked to PRs).